tests/filecheck/dialects/memref_stream/canonicalize.mlir

```diff
@@ -0,0 +1,117 @@
+// RUN: xdsl-opt -p canonicalize %s | filecheck %s
+
+// Remove unused argument
+
+// CHECK:  %E, %F, %G, %H = "test.op"() : () -> (memref<4x2xf64>, memref<2x3xf64>, memref<4x3xf64>, f64)
+%E, %F, %G, %H = "test.op"() : () -> (memref<4x2xf64>, memref<2x3xf64>, memref<4x3xf64>, f64)
+
+// CHECK-NEXT:  %I = "test.op"() : () -> memref<4x3xf64>
+%I = "test.op"() : () -> memref<4x3xf64>
+
+// Don't remove when there are uses
+// CHECK-NEXT:  memref_stream.generic {
+// CHECK-NEXT:    bounds = [4, 2, 3],
+// CHECK-NEXT:    indexing_maps = [
+// CHECK-NEXT:      affine_map<(d0, d1, d2) -> (d0, d2)>,
+// CHECK-NEXT:      affine_map<(d0, d1, d2) -> (d2, d1)>,
+// CHECK-NEXT:      affine_map<(d0, d1) -> (d0, d1)>,
+// CHECK-NEXT:      affine_map<(d0, d1) -> (d0, d1)>
+// CHECK-NEXT:    ],
+// CHECK-NEXT:    iterator_types = ["parallel", "parallel", "reduction"]
+// CHECK-NEXT:  } ins(%{{.*}}, %{{.*}} : memref<4x2xf64>, memref<2x3xf64>) outs(%{{.*}}, %{{.*}} : memref<4x3xf64>, memref<4x3xf64>) inits(%{{.*}} : f64, None) {
+// CHECK-NEXT:  ^{{.*}}(%{{.*}} : f64, %{{.*}} : f64, %{{.*}} : f64, %{{.*}} : f64):
+// CHECK-NEXT:    %{{.*}} = arith.mulf %{{.*}}, %{{.*}} : f64
+// CHECK-NEXT:    %{{.*}} = arith.addf %{{.*}}, %{{.*}} : f64
+// CHECK-NEXT:    %{{.*}} = arith.addf %{{.*}}, %{{.*}} : f64
+// CHECK-NEXT:    linalg.yield %{{.*}}, %{{.*}} : f64, f64
+// CHECK-NEXT:  }
+memref_stream.generic {
+    bounds = [4, 2, 3],
+    indexing_maps = [
+        affine_map<(d0, d1, d2) -> (d0, d2)>,
+        affine_map<(d0, d1, d2) -> (d2, d1)>,
+        affine_map<(d0, d1) -> (d0, d1)>,
+        affine_map<(d0, d1) -> (d0, d1)>
+    ],
+    iterator_types = ["parallel", "parallel", "reduction"]
+} ins(%E, %F : memref<4x2xf64>, memref<2x3xf64>) outs(%G, %I : memref<4x3xf64>, memref<4x3xf64>) inits(%H : f64, None) {
+^0(%e : f64, %f : f64, %acc_old_0 : f64, %acc_old_1 : f64):
+    %prod = arith.mulf %e, %f : f64
+    %acc_new_0 = arith.addf %acc_old_0, %prod : f64
+    %acc_new_1 = arith.addf %acc_old_1, %prod : f64
+    linalg.yield %acc_new_0, %acc_new_1 : f64, f64
+}
+
+
+// Do remove when there are no uses
+// CHECK-NEXT:  memref_stream.generic {
+// CHECK-NEXT:    bounds = [4, 2, 3],
+// CHECK-NEXT:    indexing_maps = [
+// CHECK-NEXT:      affine_map<(d0, d1, d2) -> (d0, d2)>,
+// CHECK-NEXT:      affine_map<(d0, d1) -> (d0, d1)>,
+// CHECK-NEXT:      affine_map<(d0, d1) -> (d0, d1)>
+// CHECK-NEXT:    ],
+// CHECK-NEXT:    iterator_types = ["parallel", "parallel", "reduction"]
+// CHECK-NEXT:  } ins(%{{.*}} : memref<4x2xf64>) outs(%{{.*}}, %{{.*}} : memref<4x3xf64>, memref<4x3xf64>) inits(%H : f64, None) {
+// CHECK-NEXT:  ^{{.*}}(%{{.*}} : f64, %{{.*}} : f64, %{{.*}} : f64):
```

Comment:
```suggestion
// CHECK-NEXT:  ^{{\S*}}(%{{\S*}} : f64, %{{\S*}} : f64, %{{\S*}} : f64):
```

Otherwise this will still pass for blocks with 15 arguments.